src/base/LruMap.h

```diff
@@ -135,11 +153,29 @@ LruMap<EntryValue, EntryCost>::get(const char *key)
     return NULL;
 }
 
-template <class EntryValue, size_t EntryCost>
+template <class EntryValue, size_t MemoryUsedByEV(const EntryValue *)>
 bool
-LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
+LruMap<EntryValue, MemoryUsedByEV>::add(const char *key, EntryValue *t)
 {
-    if (ttl == 0)
+    // map the retention concept into a TTL value
+
+    // do not store / retain any entries
+    if (retention == 0)
+        return false;
+
+    // negative retention means store 'forever'
+    if (retention < 0)
+        return add(key, t, 0xFFFFFFFF);
```

Comment:
Good catch. I still think the decision to use maximum storable value was best. I'll take a look and see what values we may be able to use more safely. Worst case I might have to go with dropping the magic number to 7-bit all-ones.